lib/r2d2/payment_token.rb

```diff
@@ -40,10 +40,10 @@ def generate_shared_secret(private_key, ephemeral_public_key)
         private_key.dh_compute_key(point)
       end
 
-      def derive_hkdf_keys(ephemeral_public_key, shared_secret)
-        key_material = Base64.decode64(ephemeral_public_key) + shared_secret;
-        hkdf = HKDF.new(key_material, :algorithm => 'SHA256', :info => 'Android')
-        hkdf_keys = {
+      def derive_hkdf_keys(ephemeral_public_key, shared_secret, info = 'Android')
+        key_material = Base64.decode64(ephemeral_public_key) + shared_secret
+        hkdf = HKDF.new(key_material, :algorithm => 'SHA256', :info => info)
```

Comment:
IRL may want to have the info be a constant since only a few keys are allowed. Might also be worth having another class for android 
```ruby
class AndroidPaymentToken < PaymentToken
  HKDF_INFO = 'Android'
end
```
```ruby
hkdf = HKDF.new(key_material, :algorithm => 'SHA256', :info => self.class.const_get('HKDF_INFO')
```